File: quark/test/emit/matrix/java/Matrix.java
public class Matrix<T> {
    public Integer width;
    public Integer height;
    public java.util.ArrayList<java.util.ArrayList<T>> columns;
    public Matrix(Integer width, Integer height) {
        (this).width = width;
        (this).height = height;
        this.columns = new java.util.ArrayList(java.util.Arrays.asList(new Object[]{}));
        while (((this.columns).size()) < (width)) {
            java.util.ArrayList<T> column = new java.util.ArrayList(java.util.Arrays.asList(new Object[]{}));
            while (((column).size()) < (height)) {
                (column).add(null);
            }
            (this.columns).add(column);
        }
    }
    public T __get__(Integer i, Integer j) {
        return ((this.columns).get(i)).get(j);
    }
    public void __set__(Integer i, Integer j, T value) {
        ((this.columns).get(i)).set((j), (value));
    }
}

File: quark/test/emit/matrix/py/matrix.py
from quark_runtime import *

class Matrix(object):
    def _init(self):
        self.width = None
        self.height = None
        self.columns = None

    def __init__(self, width, height):
        self._init()
        (self).width = width
        (self).height = height
        self.columns = _List([])
        while ((len(self.columns)) < (width)) :
            column = _List([]);
            while ((len(column)) < (height)) :
                (column).append(None);

            (self.columns).append(column);

    def __get__(self, i, j):
        return ((self.columns)[i])[j]

    def __set__(self, i, j, value):
        ((self.columns)[i])[j] = (value);



def main():
    matrix = Matrix(3, 3);
    (matrix).__set__(2, 2, 123);
    _println((matrix).__get__(2, 2));


if __name__ == "__main__":
    main()
